### PR TITLE
tiny-js-world-oop done

### PR DESCRIPTION
**Pull request by Temu4**
### OOP exercise

[Demo](https://temu4.github.io/a-tiny-JS-world-oop/) |
[Code base](https://github.com/Temu4/frontend-2019-p2p/tree/tiny-js-world-oop/submissions/Temu4/tiny-js-world-oop)

Redesigned according to OOP. 
Please, review.

### CODE REVIEW

**Comment by Temu4** [comment]
fixed up to Minimal requirements.

**Comment by Temu4** [comment]
> Minimal requirements to meet:
> 
> * [x]  Implement a base class to inherit from
> * [x]  Employ default parameters
> * [x]  Each species is represented with its own class
> * [x]  No need to specify species at instantiation
> * [x]  Classes for species that do not have hands by natural design
>   do not consequently have `hands` or any equivalent property
> * [x]  All inhabitants are stored in a container (array or object)
> * [x]  JS native features are intensively employed (`const`, `let`, `Array.map|join|forEach|...`, etc)
> * [x]  [OOP, SOLID and DRY](https://github.com/OleksiyRudenko/a-tiny-JS-world/blob/master/README.md#learn-on-your-own)
>   principles are intensively employed

fix

File: submissions/Temu4/tiny-js-world-oop/index.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  }
+}
+
+class Dog extends Animal {
+  constructor(gender, name, age, skinColor, hobby) {
+    super('dog', gender, name, age, skinColor, hobby);
+  }
+}
+
+const john = new Man('John', 28, 'fishing');
```
Comment (o-msh) [inline submissions/Temu4/tiny-js-world-oop/index.js]: You can also don't naming objects in this case. ` const john = ...'John'; const anny = ...'Anny' ` etc. You can define objects in array, then call `forEach` method.
```javascript
[
    new Man(...),
    new Woman(...),
].forEach(...)
```

File: submissions/Temu4/tiny-js-world-oop/index.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+const anny = new Woman('Anny', 23, 'shoping');
+const rex = new Dog('male', 'Rex', 3, 'white with black spots', 'diging bones');
```
Comment (o-msh) [inline submissions/Temu4/tiny-js-world-oop/index.js]: Better way will be use object as parameter.
```javascript
new Cat({gender: 'female', name: 'Sida' ...etc})
```
This trick will do more understand parameters.

File: submissions/Temu4/tiny-js-world-oop/index.js (hunk 1 above)
Comment (Temu4) [reply submissions/Temu4/tiny-js-world-oop/index.js]: can explain, please?

File: submissions/Temu4/tiny-js-world-oop/index.js (hunk 2 above)
Comment (Temu4) [reply submissions/Temu4/tiny-js-world-oop/index.js]: Thx for the feature! 👍  But my IDE advised which props response for which value.

File: submissions/Temu4/tiny-js-world-oop/index.js (hunk 1 above)
Comment (Temu4) [reply submissions/Temu4/tiny-js-world-oop/index.js]: fix in the last commit

File: submissions/Temu4/tiny-js-world-oop/index.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    I'm a ${this.species}, ${this.gender}. I have ${this.hands} hands and ${this.legs} legs. My hobby is ${
+      this.hobby
+    }.`;
+  }
+}
+
+class Woman extends Human {
+  constructor(name, age, hobby) {
+    super('female', name, age, hobby);
+    this.greeting = this.greeting.bind(this);
```
Comment (o-msh) [inline submissions/Temu4/tiny-js-world-oop/index.js]: Why you are using `bind` here?

File: submissions/Temu4/tiny-js-world-oop/index.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  }
+  greeting() {
```
Comment (o-msh) [inline submissions/Temu4/tiny-js-world-oop/index.js]: Your `Woman` and `Man` instance extends from basic `Human` class. Why you declare method `greeting` in two instances ? In OOP common methods must be declared in common class. You can call parent method from `super` instruction.

File: submissions/Temu4/tiny-js-world-oop/index.js (hunk 3 above)
Comment (Temu4) [reply submissions/Temu4/tiny-js-world-oop/index.js]: remove it

File: submissions/Temu4/tiny-js-world-oop/index.js (hunk 4 above)
Comment (Temu4) [reply submissions/Temu4/tiny-js-world-oop/index.js]: for `Man` and `Woman` instances with method has so many differences, that I decided to divide them.

**Pull request merged**